etcd-manager/cmd/etcd-manager/main.go

```diff
@@ -189,6 +196,32 @@ func (o *EtcdManagerOptions) InitDefaults() {
 	o.Insecure = false
 	o.EtcdInsecure = false
 	o.EtcdManagerMetricsPort = 0
+
+	o.NetworkCIDR = os.Getenv("ETCD_MANAGER_NETWORK_CIDR")
+}
+
+func parseNetworkCIDR(o *EtcdManagerOptions) ([]*net.IPNet, error) {
+	if o.NetworkCIDR == "" {
+		return nil, nil
+	}
+
+	if o.VolumeProviderID != "openstack" {
+		return nil, fmt.Errorf("is only supported with provider 'openstack'")
+	}
```

Comment:
@hakman you mentioned (in a now gone comment due to me committing the suggestion) to validate that it is only set for openstack, this should be enough?

I also test this here: https://github.com/kubernetes-sigs/etcdadm/pull/317/files#diff-6200c824e51c034077f4aaf4af4c567779591745d3f34ac7e0c53f55375a45e5R55

Reply:
I think this function should be inlined. no need to complicate things.